react-native-cli/index.js

```diff
@@ -82,17 +82,31 @@ function init(name) {
       start: 'node_modules/react-native/packager/packager.sh'
     }
   };
-  fs.writeFileSync(path.join(root, 'package.json'), JSON.stringify(packageJson));
+  fs.writeFileSync(
+    path.join(root, 'package.json'),
+    JSON.stringify(packageJson, null, 2)
+  );
   process.chdir(root);
 
-  run('npm install --save react-native', function(e) {
+  var initCmd = path.resolve(__dirname, '..', 'init.sh') + ' ' + projectName;
+  run(initCmd, function(e) {
```

Comment:
wrong order - init script assumes `node_modules/react-native` exists, but it is installed only after `npm install --save react-native`. Also I don't think you need to change this file at all, `react-native-cli` is supposed to be a simple wrapper script. Instead, revert your change to `cli.js` and instead of `init` that invokes shell, call a function on JS module